### DISCUSSION TITLE
I did a thing! A React based PWA making use of the Lanrargi API

### DISCUSSION THREAD

**Discussion by hibikikuze4dan** [Difegue/LANraragi#814] [Show and tell]
Hello! I'd like to unveil a small project I've been working on for a bit, [LRR-React-Web](https://github.com/hibikikuze4dan/lanraragi-react-web) (name is subject to change)! It is a progressive web app created using React for making use of the Lanraragi API to do things like read archives, add archives to categories, delete archives, etc.

I made to be another method of viewing archives on mobile on tablet devices (primarily Android) but it works on desktops/laptops as well. 

[Basic Image of the App](https://github.com/Difegue/LANraragi/assets/64999371/16279880-db0b-4184-bc2a-7e73387b4390)

**Comment by Difegue** [comment]
Hey this is cool, great job! I played with it a bit on desktop and it seems to have most of the base featureset. 👍  
Weirdest thing to me is that the covers don't have a fixed size and stretch depending on the screen size, idk how hard that'd be to fix?
I think this can specifically be useful for iOS devices, since people don't necessarily have the means to sideload DuReader.  

The annoying part is that you need to self-host it thru Docker, even though given it's a PWA you barely need to interact with the original host once it's installed.. Well, it is what it is. 🤷  
I'll make sure to add a link to this project in the 3rd party clients part of the Documentation. 🙏 Nice work again!

**Reply by hibikikuze4dan** [reply]
Thanks for the feedback and glad to hear you liked it! Yeah, the covers are kind of hard to get down specifically, so I just went forcing the whole thumbnail into whatever space was available for now. I'll start looking into making them work more like how they do on Lanraragi by default.

Also yeah, the Docker installation can definitely be a hurdle. I could just host the app itself on something like AWS and provide a link on LRR-React-Web's repo for users to go and get the app on their device that way, but the problem I've run into is that when making requests to the Lanraragi API from the hosted page, I end up getting errors due to the app being hosted on an https site while my Lanraragi instance is running from my NAS on an http site, and I haven't been able to find a real way around those errors.

Once again, thanks for the feedback!
